### PR TITLE
Juliacon Tutorial

### PR DESCRIPTION
**Pull request by kescobo**
Partner to https://github.com/EcoJulia/BiobakeryUtils.jl/pull/101

This isn't quite finished, but @anikaluo and @annelle-abatoni I think there's enough here to start reviewing

### CODE REVIEW

**Comment by kescobo** [comment]
Alright, this is done, I think

File: tutorials/juliacon2022.jl [hunk 1]
```diff
@@ -0,0 +1,1644 @@
...
+### 🏁 Relative abundance matrix
+
+You can use something from the `SparseArrays` standard library,
+but any old matrix will do, and `Microbiome.jl` will convert it for you.
+
+Notice that, in the DataFrame above,
+a number of taxa are missing from a number of samples.
+This is expected,
+but "missing" in the contex of microbial community data just means
+that the relative abundance is 0.
```
Comment (anikaluo) [inline tutorials/juliacon2022.jl:157]:
```suggestion
that the relative abundance is 0 (zero).
```

in Pluto formatting/font, the 0 is not very clearly a 0. maybe put the actual word zero in parentheses for clarity?

File: tutorials/juliacon2022.jl [hunk 2]
```diff
@@ -0,0 +1,1644 @@
...
+end
+
+# ╔═╡ 64624649-8a70-44fb-bbd5-5c3466ce484a
+@assert all(isfile, files)
+
+# ╔═╡ ce747551-460b-4542-b07c-7adc3fdd08cd
+md"""
+### 📊 Data format
+
+The data we've downloade here is in the form of tab-separated files,
```
Comment (anikaluo) [inline tutorials/juliacon2022.jl:79]:
```suggestion
The data we've downloaded here is in the form of tab-separated files,
```

File: tutorials/juliacon2022.jl [hunk 3]
```diff
@@ -0,0 +1,1644 @@
...
+md"""
+## Working with `CommunityProfile`s
+
+Now that we have our `CommunityProfile`, there are a number of ways to work with it.
+First, let's take a look at the various accessor functions.
+
+- `features()` - returns a vector of the features (in this case `Taxon`s)
+- `samples()` - returns a vector of the samples, including the attached metadata
+- `featurenames()`/`samplenames()` - like `features()` and `samples()`, but returns just the string representation.
+- `abundances()` - returns the sparse matrix with relative abundances`
```
Comment (anikaluo) [inline tutorials/juliacon2022.jl:289]: should there be an example of using `abundances()` below?

File: tutorials/juliacon2022.jl [hunk 4]
```diff
@@ -0,0 +1,1644 @@
...
+# ╔═╡ 95011cf6-e81b-4a48-9fdc-388df2c9ed75
+md"""
+### ℹ️ Indexing
+
+There are a number of useful ways to index into a `CommunityProfile`.
+The first and most straightforward is using numbers -
+keep in mind that indexing in julia starts at 1,
+that rows are the first dimension, and columns are the second dimension.
+
+For example, the following will return the relative abundance of the 3rd row and 6th column:
```
Comment (anikaluo) [inline tutorials/juliacon2022.jl:314]:
```suggestion
For example, the following will return the relative abundance of the 4th row and 6th column:
```

maybe?

File: tutorials/juliacon2022.jl [hunk 5]
```diff
@@ -0,0 +1,1644 @@
...
+md"""
+The `prevalence()` function calculates the fraction of samples
+that a given feature is found, also with an optional minimum abundance value:
+"""
+
+# ╔═╡ 0b80a1b8-b6c6-4663-987d-b4613536056b
+prevalence(comm, 1.0)
+
+# ╔═╡ 7fdf6e5c-471e-4170-9a7c-2c027ec02100
+md"""
```
Comment (anikaluo) [inline tutorials/juliacon2022.jl:391]: some of the code for the text sections under this heading are not hidden for me?

File: tutorials/juliacon2022.jl [hunk 6]
```diff
@@ -0,0 +1,1644 @@
...
+
+# ╔═╡ cd06d809-0a85-485e-9ec7-7698cc748344
+set!(comm, "SRS014459", :new_metadatum, "some value");
+
+# ╔═╡ 3a064eac-2e45-4565-a35b-71e3f755c5c0
+samples(comm)[1]
+
+# ╔═╡ e57c4876-f4c2-40d4-befc-1f6e82ae1496
+md"""
+Or, add lots of metadata to your community profile
```
Comment (anikaluo) [inline tutorials/juliacon2022.jl:409]: hide?

File: tutorials/juliacon2022.jl [hunk 7]
```diff
@@ -0,0 +1,1644 @@
...
+	sex    = rand(("male", "female"), 6),
+	age    = [10, 11, 12, 13, 14, 15]
+)
+
+# ╔═╡ bc0f81b3-3750-426e-8113-712f7490236e
+set!(comm, mtdt)
+
+# ╔═╡ f151ec8d-08c4-4e2c-bc2e-555b999d771a
+md"""
+And you can get all of the metadata out as a vector of `NamedTuple`s
```
Comment (anikaluo) [inline tutorials/juliacon2022.jl:426]: hide?

File: tutorials/juliacon2022.jl (hunk 3 above)
Comment (kescobo) [reply tutorials/juliacon2022.jl:289]: Nah, it's straightforward

**Pull request merged**